work_queue/src/work_queue.c

```diff
@@ -5968,7 +6059,8 @@ void work_queue_delete(struct work_queue *q)
 		struct work_queue_factory_info *f;
 		hash_table_firstkey(q->factory_table);
 		while(hash_table_nextkey(q->factory_table, &key, (void **) &f)) {
-			remove_factory(f);
+			free(f->name);
```

Comment:
Don't you need `remove_factory_info` here? It seems that the memory assigned to `f` leaks here.

Reply:
I should use `remove_factory_info`, but this will not cause mem leak. As far as I know, `hash_table_reomve` will free the memory allocated for the struct, so long as I freed the string inside the struct, this should be fine.

Reply:
It does not, a free is need.

Reply:
updated `remove_factory_info` to fix this issue.